examples/echo-nats/echo.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"fmt"
+	"github.com/nats-io/go-nats"
+	"github.com/rs/zerolog/log"
+	"regexp"
+	"strings"
+	"time"
+)
+
+type NatsClient interface {
+	QueueSubscribe(subject, queue string, cb nats.Handler) (*nats.Subscription, error)
+	Request(subject string, v interface{}, vPtr interface{}, timeout time.Duration) error
+}
+
+func HandleRequest(n NatsClient) {
+	n.QueueSubscribe("slack.event.message", "nats-echo-queue", echoHandler(n))
+}
+
+func echoHandler(n NatsClient) func(m *slackMsg) {
+	return func(m *slackMsg) {
+		if ok, _ := regexp.MatchString("^nats echo\\s+.*$", m.Text); !ok {
+			return
+		}
+		msg := slackMsg{
+			Channel:  m.Channel,
+			Text:     fmt.Sprintf("You said:\n> %s", strings.TrimPrefix(m.Text, "nats echo ")),
+			ThreadTs: m.Ts,
+		}
+
+		var resp slackPostResp
+		if err := n.Request("slack.chat.postMessage", &msg, &resp, 10*time.Second); err != nil || resp.Error != "" {
+			log.Error().
+				Err(err).
+				Str("resp", fmt.Sprintf("%+v", resp)).
+				Str("subject", "slack.chat.postMessage").
+				Str("message", m.Text).
+				Msg("error publishing message to NATS")
+		}
+	}
+}
+
+type slackMsg struct {
```

Comment:
Nothing wrong with this however maybe you can just use slack's message struct from `github.com/nlopes/slack` this is what slack-nats will be broadcasting similar for the putput message

Reply:
Wouldn't do that - no need to import a whole library package (and its dependencies), just for the few fields you want to deserialise

Reply:
As mentioned nothing wrong with this.

The main reasons I've mentioned it are:
- this is what slack nats is broadcasting and this is what we will receive over the wire (this is where the main penalty is)
- now we need to maintain this struct(s)
- every other project might have a copy of this struct(s)
- not sure what the penalty for unmarshalling is and it could be considered premature optimisation

Having said that I've already approved PR and not fussed either way :)